src/wallet/coin_selection.rs

```diff
@@ -66,10 +67,25 @@
 //!             });
 //!         }
 //!
-//!         Ok(CoinSelectionResult {
-//!             selected: all_utxos_selected,
-//!             fee_amount: fee_amount + additional_fees,
-//!         })
+//!         let calculated_waste = Some(
+//!             calculate_waste(
+//!                 required_utxos
+//!                     .into_iter()
+//!                     .chain(optional_utxos)
+//!                     .collect::<Vec<WeightedUtxo>>(),
```

Comment:
Yes, that's conceptually correct but infesable with the actual method's
signature. Currently is not possible to consume `Vec<Utxo>` for
`calculate_waste`, just `Vec<WeightedUtxo>`. That might change also, if we
change the `selected` parameter by something else.
I suggest to finish the `calculate_waste` function signature first and then
apply this kind of changes. It may require some modifications to the
`AlwaysSpendEverything` algorithm outside of the scope modified by
`calculate_waste`, are you OK with that?

Reply:
Fixed with [d2bf84a](https://github.com/bitcoindevkit/bdk/pull/558/commits/d2bf84af45283e9b896788b74383c805f110cf4b).